addons/mail/static/src/models/sender/sender.js

```diff
@@ -0,0 +1,141 @@
+                    });
+                    if (!composer.isLog) {
+                        params.context = { mail_post_autofollow: true };
+                    }
+                }
+                if (composition.composerView.threadView && composition.composerView.threadView.replyingToMessageView && composition.composerView.threadView.thread !== this.messaging.inbox) {
+                    postData.parent_id = composition.composerView.threadView.replyingToMessageView.message.id;
+                }
+                const { threadView = {} } = composition.composerView;
+                const { thread: chatterThread } = composition.composerView.chatter || {};
+                const { thread: threadViewThread } = threadView;
+                const messageData = await this.env.services.rpc({ route: `/mail/message/post`, params });
+                if (!this.messaging) {
+                    return;
+                }
+                const composerView = composition.composerView;
+
+                const messageConverted = this.messaging.models['Message'].convertData(messageData);
+                this.messaging.modelManager._delete(composition.message);
+                composition.update({
+                    message: insertAndReplace(messageConverted),
+                });
+
+                for (const threadView of composition.message.originThread.threadViews) {
+                    // Reset auto scroll to be able to see the newly posted message.
+                    threadView.update({ hasAutoScrollOnMessageReceived: true });
+                }
+                if (chatterThread) {
+                    if (chatterThread.exists()) {
+                        chatterThread.fetchData(['followers', 'suggestedRecipients']);
+                    }
+                }
+                if (threadViewThread) {
+                    if (threadViewThread === this.messaging.inbox) {
+                        if (composerView) {
+                            composerView.delete();
+                        }
+                        this.env.services['notification'].notify({
+                            message: _.str.sprintf(this.env._t(`Message posted on "%s"`), composition.message.originThread.displayName),
+                            type: 'info',
+                        });
+                    }
+                    if (threadView && threadView.exists()) {
+                        threadView.update({ replyingToMessageView: clear() });
+                    }
+                }
+                composition.delete();
+            } catch(error) {
+                this.update({
+                    isSending: false,
+                });
+                this.watchQueue();
+            }
+        },
+        watchQueue() {
+            if (!this.isSending && this.messagesToSend.length > 0) {
+                const interval = 5000;
+                setInterval(this._processQueue, interval);
```

Comment:
Should properly clear the interval when needed, at least on deletion of the record.

Reply:
This part of the code is WIP/Experimental.